app/controllers/works_controller.rb

```diff
@@ -183,7 +183,11 @@ def after_save(form:, event_context:)
 
     # Instead of crazy conditional logic, constructing the event name and sending
     state_event = state_event_for(work_version)
-    work_version.send(state_event) if state_event
+    if state_event
+      work_version.send(state_event)
+      FetchGlobusJob.perform_later(work_version) if state_event.starts_with?("fetch_globus")
+      UnzipJob.perform_later(work_version) if state_event.starts_with?("unzip")
+    end
```

Comment:
With the job enqueued here, I haven't been able to trigger any invalid transition errors. I think it *might* still be possible for the job to finish and send a job completed event before the state machine has saved the new state in the database, but I haven't seen that happen yet. Open to other ideas about how to do this.